.rubocop.yml

```diff
@@ -0,0 +1,67 @@
+AllCops:
+  TargetRubyVersion: '2.7.6'
+  Exclude:
+    - Rakefile
+    - "rakelib/**/*"
+    - "**/example/*"
+    - "bin/instrumentation_generator"
+    - "**/**/vendor/bundle/**/*"
+Bundler/OrderedGems:
+  Enabled: false
+Gemspec/RequiredRubyVersion:
+  Enabled: false
+Metrics/AbcSize:
+  Enabled: false
+Metrics/BlockLength:
+  Enabled: false
+Metrics/MethodLength:
+  Max: 20
+Lint/ConstantDefinitionInBlock:
+  Exclude:
+    - "**/*test.rb"
+Lint/MixedRegexpCaptureTypes:
+  Exclude:
+    - propagator/xray/lib/opentelemetry/propagator/xray/text_map_propagator.rb
+Lint/UnusedMethodArgument:
+  Enabled: false
+Naming/FileName:
+  Exclude:
+    - '**/opentelemetry-*'
+Naming/MethodParameterName:
+  Exclude:
+    - instrumentation/dalli/lib/opentelemetry/instrumentation/dalli/patches/server.rb
```

Comment:
We currently have a mix of disabling specific rules inline when we do not want them to apply to all files so I am not sure how I feel about overrides for individual files being moved into the parent Rubocop file. 

I know they probably already existed in the gem specific configs but IMO we should either disable these rules in individual source files or override them closer to a gem specific Rubocop config.

Reply:
Yeah, great point. I'm for moving all exceptions to specific files. I can do a pass at that.